### DISCUSSION TITLE
Server Side Rendering

### DISCUSSION THREAD

**Discussion by NemuruYama** [AvaloniaUI/Avalonia#17733] [Ideas]
Hello,

I want to try using Avalonia to have an application that can both be a website and a desktop application (basically opposite of how Electron works) and I was wondering if there are any possibilities for initial SSR rendering, like Angular/Blazor have.

It would probably help with the initial load, for the web version, and I don't know how difficult it is to implement.

**Comment by stevemonaco** [comment]
Avalonia doesn't use HTML, so SSR seems a moot point. For Avalonia running on browser, there is a small amount of HTML/JS to bootstrap, but the app itself is WASM which is rendered by Skia onto a canvas without any specific DOM elements.

**Comment by thevortexcloud** [comment]
If you need it 100% server side (meaning no app code ever reaches the client), Avalonia does have a VNC backend. You could just hook a VNC client up to it in a browser and on a desktop.
